services/src/workflows/postgres_workflow_registry.rs

```diff
@@ -0,0 +1,57 @@
+use crate::contexts::PostgresDb;
+use crate::error::Result;
+use crate::workflows::workflow::{Workflow, WorkflowId};
+use crate::{error, workflows::registry::WorkflowRegistry};
+use async_trait::async_trait;
+use bb8_postgres::{
+    tokio_postgres::tls::MakeTlsConnect, tokio_postgres::tls::TlsConnect, tokio_postgres::Socket,
+};
+use snafu::ResultExt;
+
+#[async_trait]
+impl<Tls> WorkflowRegistry for PostgresDb<Tls>
+where
+    Tls: MakeTlsConnect<Socket> + Clone + Send + Sync + 'static,
+    <Tls as MakeTlsConnect<Socket>>::Stream: Send + Sync,
+    <Tls as MakeTlsConnect<Socket>>::TlsConnect: Send,
+    <<Tls as MakeTlsConnect<Socket>>::TlsConnect as TlsConnect<Socket>>::Future: Send,
+{
+    async fn register_workflow(&self, workflow: Workflow) -> Result<WorkflowId> {
+        let conn = self.conn_pool.get().await?;
+        let stmt = conn
+            .prepare(
+                "INSERT INTO workflows (id, workflow) VALUES ($1, $2) 
+            ON CONFLICT DO NOTHING;",
```

Comment:
if a workflow with the id already exists

Reply:
But shouldn't we then throw an error instead of ignoring it?

Reply:
The idea was to have an "INSERT IF NOT EXISTS". Otherwise we would need two operations, first to check if it already exists, and seconds to insert it. IMHO we do not want an error here, as multiple users can create the same workflow independently of each other.